### PR TITLE
Added TS Definitions

### PR DESCRIPTION
**Pull request by D4RKAR117**
**This tries to solve issue #95** 

Recommended review by @SqrtMinusOne

### CODE REVIEW

**Comment by SqrtMinusOne** [comment]
The [should also be](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) a `type` property in package.json:
```json
  "types": "./types/index.d.ts",
```
After that, the package can be imported.

However, there is no type-checking for props; I'm not sure how to make a proper model augmentation for Vue components, perhaps I'll take a look in a couple of days.

**Comment by D4RKAR117** [comment]
@SqrtMinusOne I missed that little detail, thanks for identifying it. I think it is the basis to be complemented and will provide good compatibility with Typescript, thanks to you.

**Comment by D4RKAR117** [comment]
I just added the **Types** flag into the package.json

**Comment by yaminncco** [comment]
@D4RKAR117 thanks I will look at this soon

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
+declare module "vue-sidebar-menu" {
+  import Vue from "vue";
```
Comment (yaminncco) [inline types/index.d.ts]:
```suggestion
  import Vue, { PluginObject } from "vue";
```

File: types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    theme?: string;
+
+    /**
+     * Disable hover on collapse mode.
+     * by default false.
+     */
+    disableHover?: boolean;
+  }
+  export { SidebarMenu };
+  export default SidebarMenu;
```
Comment (yaminncco) [inline types/index.d.ts]:
```suggestion
  const VueSidebarMenu: PluginObject<any>;
  export default VueSidebarMenu;
```

File: package.json [hunk 3]
```diff
@@ -27,7 +28,7 @@
     "dev": "cross-env NODE_ENV=development webpack-dev-server --hot --open --config webpack.config.dev.js",
     "build:demo": "cross-env NODE_ENV=production webpack --progress --hide-modules --config webpack.config.dev.js",
     "build:lib": "cross-env NODE_ENV=production webpack --progress --hide-modules --config webpack.config.prod.js",
-    "lint": "eslint --ext .js,.vue ."
+    "lint": "eslint --ext .js,.vue,.ts ."
```
Comment (yaminncco) [inline package.json]: need to install eslint typescript parser
```suggestion
    "lint": "eslint --ext .js,.vue ."
```

**Comment by D4RKAR117** [comment]
@yaminncco Now it should be fine, or so I think, I never thought about working as a plugin, I always use it as a local component, it's good to know

**Pull request merged**